proxy_ring_buffer.go

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package hazelcast
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	hzcerrors "github.com/hazelcast/hazelcast-go-client/internal/hzerrors"
+	"github.com/hazelcast/hazelcast-go-client/internal/proto/codec"
+	iserialization "github.com/hazelcast/hazelcast-go-client/internal/serialization"
+)
+
+// A Ringbuffer is a data structure where the content is stored in a ring-like
+// structure. A Ringbuffer has a fixed capacity, so it won't grow beyond
+// that capacity and endanger the stability of the system. If that capacity
+// is exceeded, the oldest item in the Ringbuffer is overwritten.
```

Comment:
We try to put each sentence on its own line, so each line ends with a period. Would it be possible for you to adjust this documentation and others for that convention?

Reply:
Done.
9aacf8e0b5617979552f848faa1f61b660b73f4a